Mutagen.Bethesda.Analyzers/IAnalyzerRunner.cs

```diff
@@ -1,222 +1,85 @@
-﻿using System.IO.Abstractions;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
 using Autofac;
-using Mutagen.Bethesda.Analyzers.Autofac;
-using Mutagen.Bethesda.Analyzers.Config.Run;
-using Mutagen.Bethesda.Analyzers.Config.Topic;
-using Mutagen.Bethesda.Analyzers.Drivers;
-using Mutagen.Bethesda.Analyzers.Reporting.Drops;
+using Mutagen.Bethesda.Analyzers.Engines;
 using Mutagen.Bethesda.Analyzers.Reporting.Handlers;
-using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
-using Mutagen.Bethesda.Analyzers.SDK.Caches;
 using Mutagen.Bethesda.Analyzers.SDK.Drops;
 using Mutagen.Bethesda.Analyzers.SDK.Topics;
-using Mutagen.Bethesda.Analyzers.Skyrim;
-using Mutagen.Bethesda.Environments.DI;
 using Mutagen.Bethesda.Plugins;
-using Mutagen.Bethesda.Plugins.Cache;
-using Mutagen.Bethesda.Plugins.Meta;
-using Mutagen.Bethesda.Plugins.Order;
 using Mutagen.Bethesda.Plugins.Records;
-using Noggog.WorkEngine;
 
 namespace Mutagen.Bethesda.Analyzers;
 
 public interface IAnalyzerRunner
 {
     /// <summary>
-    /// Analyze a mod for issues.
+    /// Run the analysis
     /// </summary>
-    /// <param name="mod">Mod to analyze</param>
-    /// <returns>Analysis results for topics found in the mod </returns>
-    Task<IEnumerable<AnalyzerResult>> Analyze(IModGetter mod);
-
-    /// <summary>
-    /// Analyze a major record for issues.
-    /// </summary>
-    /// <param name="record">Major record to analyze</param>
-    /// <returns>Analysis results for topics found in the record</returns>
-    Task<IEnumerable<AnalyzerResult>> Analyze<TMajorRecord>(TMajorRecord record)
-        where TMajorRecord : IMajorRecordGetter;
+    /// <returns>Analysis results for topics found in the run</returns>
+    IAsyncEnumerable<AnalyzerResult> Analyze();
```

Comment:
I think Analyze(mod) or Analyze(Record) would still be quite useful, because in a scenario where you want to analyze like these 100 records, you'd have to create 100 analyzer runners atm right?

Reply:
https://github.com/Mutagen-Modding/Mutagen.Bethesda.Analyzers/issues/508
https://github.com/Mutagen-Modding/Mutagen.Bethesda.Analyzers/issues/509
https://github.com/Mutagen-Modding/Mutagen.Bethesda.Analyzers/issues/510